### DISCUSSION TITLE
Add Login Page for Multiple User Permission before MkDocs

### DISCUSSION THREAD

**Discussion by mangeldev13** [squidfunk/mkdocs-material#5360] [Q&A]
I want to add log in page before anyone can open my MkDocs. 
The requirement:
There are 2 user permission
   - 1 as admin, can edit and read
   - 1 as reader, only to read

What should I do?

**Comment by berkantsahn** [comment]
I don't know much about this subject but i found something like this. You can check this repo if you'd like: https://github.com/otsuarez/mkdocs_auth

**Reply by squidfunk** [reply]
> This solution adds basic authentication the pages generated by mkdocs. It renames static html files as php files and adds php code which provides the basic authentication functionality.

I absolutely do not recommend to use this. It's a huge security risk and won't work with GitHub Pages. For basic authentication, you can use _any_ web server like nginx or Apache. Please don't forget that MkDocs is only a static site generator, not a web server. You can host the built site with any technology. That's the whole point of a static site 😉

**Reply by berkantsahn** [reply]
Yes, actually a login page can be written from scratch and mkdocs can embedded in it. I came across something like this as a result of a quick search, but I was not aware that it would create a security vulnerability.

Then it makes more sense to create a login page and then embed the static site in it.

**Reply by mangeldev13** [reply]
Hii.. so after I edit my mkdoccs,  run `mkdocs build`, and I can put the site folder on xampp. Then I can add log in page which connect to database?

**Reply by mangeldev13** [reply]
> Please don't forget that MkDocs is only a static site generator, not a web server. You can host the built site with any technology. That's the whole point of a static site 😉

I have another thing that confused me. after I run `mkdocs build`. Then how can I edit the mkdocs file? For example, if I have something that I need to edit on page Home, do I have to run `mkdocs build` again then replace the old files before?

Or if I want to add another page, can I just add that file on my site folder?

**Reply by squidfunk** [reply]
> For example, if I have something that I need to edit on page Home, do I have to run mkdocs build again then replace the old files before?

Yes, correct. If you change any file inside your `docs` dir or `mkdocs.yml`, run `mkdocs build` again.

**Reply by mangeldev13** [reply]
> Yes, correct. If you change any file inside your `docs` dir or `mkdocs.yml`, run `mkdocs build` again.

Thanks for your information
